module/mempool/stdmap/backend.go

```diff
@@ -37,111 +35,107 @@ func NewBackend(options ...OptionFunc) *Backend {
 }
 
 // Has checks if we already contain the item with the given hash.
-func (b *Backend) Has(entityID flow.Identifier) bool {
+func (b *Backend[K, V]) Has(key K) bool {
 	// bs1 := binstat.EnterTime(binstat.BinStdmap + ".r_lock.(Backend)Has")
 	b.RLock()
 	// binstat.Leave(bs1)
 
 	// bs2 := binstat.EnterTime(binstat.BinStdmap + ".inlock.(Backend)Has")
 	// defer binstat.Leave(bs2)
 	defer b.RUnlock()
-	has := b.mutableBackData.Has(entityID)
+	has := b.mutableBackData.Has(key)
 	return has
 }
 
 // Add adds the given item to the pool.
-func (b *Backend) Add(entity flow.Entity) bool {
-	// bs0 := binstat.EnterTime(binstat.BinStdmap + ".<<lock.(Backend)Add")
-	entityID := entity.ID() // this expensive operation done OUTSIDE of lock :-)
-	// binstat.Leave(bs0)
-
+func (b *Backend[K, V]) Add(key K, value V) bool {
 	// bs1 := binstat.EnterTime(binstat.BinStdmap + ".w_lock.(Backend)Add")
 	b.Lock()
 	// binstat.Leave(bs1)
 
 	// bs2 := binstat.EnterTime(binstat.BinStdmap + ".inlock.(Backend)Add")
 	// defer binstat.Leave(bs2)
 	defer b.Unlock()
-	added := b.mutableBackData.Add(entityID, entity)
+	added := b.mutableBackData.Add(key, value)
 	b.reduce()
 	return added
 }
 
 // Remove will remove the item with the given hash.
```

Comment:
```suggestion
// Remove removes the value with the given key.
// If the key-value pair exists, returns the value and true.
// Otherwise returns the zero value for type V and false.
```